runtime/src/main/cpp/Memory.h

```diff
@@ -561,27 +509,73 @@ void EnsureNeverFrozen(ObjHeader* obj);
 }
 #endif
 
+
+struct FrameOverlay {
+  void* arena;
+  FrameOverlay* previous;
+  // As they go in pair, sizeof(FrameOverlay) % sizeof(void*) == 0 is always held.
+  int32_t parameters;
+  int32_t count;
+};
+
 // Class holding reference to an object, holding object during C++ scope.
 class ObjHolder {
  public:
-   ObjHolder() : obj_(nullptr) {}
+   ObjHolder() : obj_(nullptr) {
```

Comment:
So it is actually now more like `ObjStackHolder`, `StackRefHolder` or even `StackFrame`.